pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,278 @@
+package client
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	mocks "github.com/Azure/aks-tls-bootstrap-client/pkg/client/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var _ = Describe("TLS Bootstrap kubeconfig tests", func() {
+	var (
+		mockCtrl               *gomock.Controller
+		imdsClient             *mocks.MockImdsClient
+		aadClient              *mocks.MockAadClient
+		tlsBootstrapClient     *tlsBootstrapClientImpl
+		mockReader             *mocks.MockfileReader
+		tempKubeConfigName     string
+		tempKubeConfigLocation string
+	)
+
+	Context("isKubeConfigStillValid Tests", func() {
+		BeforeEach(func() {
+			tempKubeConfigName = "dummykubeconfig"
```

Comment:
nit: set this values once outside of the BeforeEach since they're always the same?